src/test/scala/nl.knaw.dans.easy.deposit/StateManagerSpec.scala

```diff
@@ -0,0 +1,165 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.deposit
+
+import better.files.File
+import nl.knaw.dans.easy.deposit.Errors.{ IllegalStateTransitionException, PropertyNotFoundException }
+import nl.knaw.dans.easy.deposit.docs.StateInfo
+import nl.knaw.dans.easy.deposit.docs.StateInfo.State
+
+import scala.util.{ Failure, Success }
+
+class StateManagerSpec extends TestSupportFixture {
+  private val draftDeposit: File = testDir / "draft"
+  private val submitBase = testDir / "submitted"
+
+  private def submittedPropsFile = (submitBase / uuid.toString).createDirectories() / "deposit.properties"
+
+  private def draftPropsFile = draftDeposit.createDirectories() / "deposit.properties"
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    clearTestDir()
+  }
+
+  "getStateInfo" should "not access submitted properties for state DRAFT" in {
+    val message = "Deposit is open for changes."
+    draftPropsFile.writeText(
+      s"""state.label = DRAFT
+         |state.description = $message
+      """.stripMargin)
+    StateManager(draftDeposit, File("does-not-exist")).getStateInfo should matchPattern {
+      case Success(StateInfo(State.draft, `message`)) =>
+    }
+  }
+
+  it should "require a bag-store.bag-id for state SUBMITTED" in {
+    draftPropsFile.writeText(
+      s"""state.label = SUBMITTED
+         |state.description = The dataset is ready for processing
+      """.stripMargin)
+    StateManager(draftDeposit, File("does-not-exist")).getStateInfo should matchPattern {
+      case Failure(e: PropertyNotFoundException) if e.getMessage ==
+        s"'bag-store.bag-id' not found in $draftDeposit/deposit.properties" =>
+    }
+  }
+
+  it should "require submit properties for state SUBMITTED" in {
+    draftPropsFile.writeText(
+      s"""state.label = SUBMITTED
+         |state.description = The dataset is ready for processing
+         |bag-store.bag-id = $uuid
+      """.stripMargin)
+    StateManager(draftDeposit, testDir / "does-not-exist").getStateInfo should matchPattern {
+      case Failure(e: PropertyNotFoundException) if e.getMessage ==
+        s"'state.label' not found in $testDir/does-not-exist/$uuid/deposit.properties" =>
+      // actually the submitted deposit does not exist
+      // for example due to lack of space to create a copy in the staging area
+      // or removed by some ignorant clean-up action
```

Comment:
How to alert the curators/application-manager to check what applies and what to do? For example set the state to rejected to resubmit, change the state to archived or whatever applies.